src/test/java/duke/util/ParserTest.java

```diff
@@ -0,0 +1,174 @@
+package duke.util;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import duke.exception.DukeException;
+import duke.exception.InvalidTaskDeletionException;
+import duke.exception.InvalidTaskDoneException;
+import duke.exception.NoDescriptionAndTimeException;
+import duke.exception.NoDescriptionException;
+import duke.exception.NoTimeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.Todo;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import org.junit.jupiter.api.Test;
+
+public class ParserTest {
```

Comment:
I like how comprehensive these unit tests are.

Reply:
Thank you!!